src/makeRobot.js

```diff
@@ -38,7 +38,92 @@
  * @return {object}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name: name,
+    chipVersion: version,
+    wheels: wheels,
+    coords: {
+      x: 0, y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, chip version: `
+              + `${this.chipVersion}, wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(value) {
+      if (typeof value === 'undefined') {
+        this.coords.y += 1;
+
+        return this;
+      }
+
+      if (value <= 0) {
+        return this;
+      }
+
+      this.coords.y += value;
+
+      return this;
+    },
+
+    goBack(value) {
+      if (typeof value === 'undefined') {
+        this.coords.y -= 1;
+
+        return this;
+      }
+
+      if (value <= 0) {
+        return this;
+      }
+
+      this.coords.y -= value;
+
+      return this;
+    },
```

Comment:
Думаю методы движения можно сократить
```suggestion
    goBack(value = 1) {
      if (value >= 1) {
        this.coords.y -= numberOfSteps;
      }

      return this;
    },
```

Reply:
Спасибо, я даже не подумал про аргументы функций по умолчанию О_о